[DPE-7244] Add landing pages (#352)
canonical/kafka-operator: docs/how-to/upgrade.md

(how-to-upgrade)=
# How to upgrade between versions

This guide applies for in-place upgrades that involve (at most) minor version upgrade of Apache Kafka workload, e.g. between Apache Kafka 3.4.x to 3.5.x.

```{warning}
Major workload upgrades are generally **NOT SUPPORTED**. 
They should be carried out using [full cluster-to-cluster migrations](how-to-cluster-replication-migrate-a-cluster).
```

```{note}
This feature is available on Charmed Apache Kafka and Charmed Apache ZooKeeper from revisions 134 and 103, respectively. Upgrade from previous versions is **not supported**, although possible (see [example](https://github.com/deusebio/kafka-pre-upgrade-patch)).
```

> Charm upgrades can include both upgrades of operator code (e.g. the revision used by the charm) and/or the workload version. Note that since the charm code pins a particular version of the workload, a charm upgrade may or may not involve also a workload version upgrade.

Author: ```suggestion
Charm upgrades can include both upgrades of charmed operator code (e.g., the revision used by the charm) and/or the workload version. Note that since the charm code pins a particular version of the workload, a charm upgrade can also involve a minor workload version upgrade.
```

(how-to-upgrade)=

> # How to upgrade between versions

Reviewer: If we drop `minor` from the title, the FIRST thing in the doc needs to be a 'warning' or whatever rich text-box you prefer, saying plainly what our supported upgrade paths are.

> # How to upgrade between versions

Author: I've improved the warnings in https://github.com/canonical/kafka-operator/pull/352/commits/0fd97a67c66764da62aeefa2f89d84ce0947b935
